include/Transforms/ListenerCSE.h

```diff
@@ -0,0 +1,23 @@
+//===-- ListenerCSE.h - Common subexpr elimination with a listener --------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_IREE_SANDBOX_TRANSFORMS_LISTENERCSE_H
+#define LLVM_IREE_SANDBOX_TRANSFORMS_LISTENERCSE_H
+
+#include "Transforms/Listener.h"
+#include "mlir/IR/Operation.h"
```

Comment:
Nit: could we forward-declare Operation as well? We are only using it as a pointer in signature.